src/main.ts

```diff
@@ -1,34 +1,76 @@
 import * as temp from 'temp'
+import * as os from 'os'
 
 import * as artifact from '@actions/artifact'
 import * as core from '@actions/core'
 import * as exec from '@actions/exec'
 import * as glob from '@actions/glob'
 import * as io from '@actions/io'
 
+class ExecOptions{
+  public listeners: object = {}
+}
+
+// Maybe (if we can get it to work) we have a param for tacking on a new error
+// message
+async function execWrapper(commandLine: string,
+                           args?: string[]): Promise<number> {
+    let myOutput = ''
+    let myError = ''
+
+    let options = new ExecOptions
+    options.listeners = {
+      stdout: (data: Buffer) => {
+        myOutput += data.toString()
+      },
+      stderr: (data: Buffer) => {
+        myError += data.toString()
+      }
+    }
+
+    return await exec.exec(commandLine, args, options)
+}
+
 async function main(): Promise<void> {
   try {
     const homeDir: string | undefined = process.env.HOME
     const buildDir = `${homeDir}/built-package`
     const minicondaDir = `${homeDir}/miniconda`
     const minicondaBinDir = `${minicondaDir}/bin`
-    const channels = '-c qiime2-staging/label/r2020.6 -c conda-forge -c bioconda -c defaults'
 
     core.addPath(minicondaBinDir);
 
     // TODO: fix these hacks
     core.addPath('../../_actions/qiime2/action-library-packaging/alpha1')
     core.addPath('.')
 
-    const installMinicondaScript: string = await io.which('install_miniconda.sh', true)
-    const installMinicondaExitCode = await exec.exec(`sh ${installMinicondaScript}`, [minicondaDir])
+    let condaURL = ''
+    if (os.platform() === 'linux') {
+      condaURL = 'https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh'
+    } else if (os.platform() === 'darwin' ) {
+      condaURL = 'https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh'
+    } else {
+      throw Error('Unsupported OS, must be Linux or Mac')
+    }
+
+    await execWrapper('wget', ['-O', 'miniconda.sh', condaURL])
+    await execWrapper('chmod', ['+x', 'miniconda.sh'])
+
+    await execWrapper('./miniconda.sh', ['-b', '-p', minicondaDir])
+
+    await execWrapper('conda', ['upgrade', '-n', 'base', '-q', '-y', '-c', 'defaults', '--override-channels', 'conda'])
```

Comment:
Can you factor this out into a new util function, something like `installMiniconda`?